tests/models/test_layout.py

```diff
@@ -19,25 +19,198 @@
+
+
+  def test_set_relative_expiration(self):
+    """Test adding expiration date relative from today. """
+    layout = Layout()
+    with self.assertRaises(securesystemslib.exceptions.FormatError):
+      layout.set_relative_expiration(days=None, months=0, years=0)
+
+    with self.assertRaises(securesystemslib.exceptions.FormatError):
+      layout.set_relative_expiration(days=0, months="", years=0)
+
+    with self.assertRaises(securesystemslib.exceptions.FormatError):
+      layout.set_relative_expiration(days=0, months=0, years=[])
+
+    layout.set_relative_expiration(days=1)
+    layout._validate_expires()
+    layout.set_relative_expiration(months=2)
+    layout._validate_expires()
+    layout.set_relative_expiration(years=3)
+    layout._validate_expires()
+    layout.set_relative_expiration(days=3, months=2, years=1)
+    layout._validate_expires()
+
+
+  def test_get_step_name_list(self):
+    """Test getting list of step names. """
+    names = ["a", "b", "c"]
+    layout = Layout(steps=[Step(name=name) for name in names])
+    self.assertListEqual(layout.get_step_name_list(), names)
+
+
+  def test_get_step_by_name(self):
+    """Test getting step by name. """
+    names = ["a", "b", "c"]
+    layout = Layout(steps=[Step(name=name) for name in names])
+    self.assertEqual(layout.get_step_by_name("b").name, "b")
+
+    with self.assertRaises(securesystemslib.exceptions.FormatError):
+      layout.get_step_by_name(None)
+
+
+  def test_remove_step_by_name(self):
+    """Test removing step by name. """
+    names = ["a", "b", "c"]
+    layout = Layout(steps=[Step(name=name) for name in names])
+    layout.remove_step_by_name("b")
+    self.assertEqual(len(layout.steps), 2)
+    self.assertTrue("b" not in layout.get_step_name_list())
+
+    with self.assertRaises(securesystemslib.exceptions.FormatError):
+      layout.get_step_by_name([])
+
+
+  def test_get_inspection_name_list(self):
+    """Test getting list of inspection names. """
+    names = ["a", "b", "c"]
+    layout = Layout(inspect=[Inspection(name=name) for name in names])
+    self.assertListEqual(layout.get_inspection_name_list(), names)
+
+
+  def test_get_inspection_by_name(self):
+    """Test getting inspection by name. """
+    names = ["a", "b", "c"]
+    layout = Layout(inspect=[Inspection(name=name) for name in names])
+    self.assertEqual(layout.get_inspection_by_name("b").name, "b")
+
+    with self.assertRaises(securesystemslib.exceptions.FormatError):
+      layout.get_inspection_by_name(1)
+
+
+  def test_remove_inspection_by_name(self):
+    """Test removing inspection by name. """
+    names = ["a", "b", "c"]
+    layout = Layout(inspect=[Inspection(name=name) for name in names])
+    layout.remove_inspection_by_name("b")
+    self.assertEqual(len(layout.inspect), 2)
+    self.assertTrue("b" not in layout.get_inspection_name_list())
+
+    with self.assertRaises(securesystemslib.exceptions.FormatError):
+      layout.remove_inspection_by_name(False)
```

Comment:
I wonder if testing the following sequence would be a good idea:

1. Add step `a`
1. Remove step `a`
1. Add *inspection* `a`
1. Remove inspection `a`

Reply:
Why exactly? Note, I didn't add `add_{step, inspection}` methods in this PR, which need testing. Currently the remove methods are tested separately like so:
```
for item in [step, inspection]:
    create layout with items a, b, c
    remove item a
    assert that item is not on on layout
```
However, I can add an assert that the item is on the layout before being removed.